app/lib/efile/az/az140_calculator.rb

```diff
@@ -77,6 +79,25 @@ def calculate_line_19
         line_or_zero(:AZ140_LINE_14)
       end
 
+      def calculate_line_29A
+        # total subtraction amount for pensions up to the maximum of $2,500 each for primary and spouse
+        return 0 unless @intake.primary_received_pension_yes? || @intake.spouse_received_pension_yes?
+
+        primary_pension_amt = @intake.primary_received_pension_yes? && @intake.primary_received_pension_amount.present? ? [@intake.primary_received_pension_amount&.round, 2_500].min : 0
```

Comment:
we don't need the ampersand here or on # 91, right? because we just checked to see if it was present, and we can't remove the .present? check because then the .min function might break. 

there's a lot happening on this line so it would be my preference to break it out into an if/else, or have two lines (one to get the pension amt and one to get the subtraction amt). 

a tiny case could also be made to also extract this to a method to be used for both primary and spouse, which normally I wouldn't bother with, but it would mean keeping the 2_500 value in one place. up to you though

Reply:
Mike and I paired to refactor this line, happy to discuss the changes if you're interested